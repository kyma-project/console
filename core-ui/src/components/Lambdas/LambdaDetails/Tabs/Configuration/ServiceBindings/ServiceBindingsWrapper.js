import React from 'react';
import LuigiClient from '@kyma-project/luigi-client';
import PropTypes from 'prop-types';

import { useQuery } from '@apollo/react-hooks';
import { GET_SERVICE_INSTANCES } from '../../../../../../gql/queries';

import { Spinner } from 'react-shared';

import { ServiceBindingsService } from './ServiceBindingsService';
import ServiceBindings from './ServiceBindings';

import { filterServiceInstances } from './helpers';

export default function ServiceBindingsWrapper({ lambdaName }) {
  const namespace = LuigiClient.getEventData().environmentId;
  const { data, error, loading, refetch } = useQuery(GET_SERVICE_INSTANCES, {
    variables: {
      namespace,
    },
    fetchPolicy: 'no-cache',
  });

  if (error) {
    return `Error! ${error.message}`;
  }
  if (loading) {
    return <Spinner />;
  }

  const [
    injectedServiceInstances,
    notInjectedServiceInstances,
  ] = filterServiceInstances(lambdaName, data.serviceInstances);

  return (
    <ServiceBindingsService lambdaName={lambdaName}>
      <ServiceBindings
        injectedServiceInstances={injectedServiceInstances}
        notInjectedServiceInstances={notInjectedServiceInstances}
        refetchQuery={refetch}
      />
    </ServiceBindingsService>
  );
}

ServiceBindingsWrapper.propTypes = {
  lambdaName: PropTypes.string.isRequired,
};
