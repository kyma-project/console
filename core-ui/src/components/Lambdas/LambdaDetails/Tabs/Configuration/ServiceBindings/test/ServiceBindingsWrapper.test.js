import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MockedProvider } from '@apollo/react-testing';
import { Spinner } from 'react-shared';

import { lambda } from './gqlMocks';
import { componentUpdate } from '../../../../../../../testing';
import ServiceBindingsWrapper from '../ServiceBindingsWrapper';

jest.mock('@kyma-project/luigi-client', () => {
  return {
    getEventData: () => ({ environmentId: 'testnamespace' }),
  };
});

describe('ServiceBindingsWrapper', () => {
  it('Renders with an error in no lambdaName is provided', async () => {
    console.error = jest.fn();
    let component;
    await renderer.act(async () => {
      component = renderer.create(
        <MockedProvider>
          <ServiceBindingsWrapper />
        </MockedProvider>,
      );
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(console.error.mock.calls.length).toBe(1);
    expect(console.error.mock.calls[0][0]).toMatchSnapshot();
  });

  it('Shows loading indicator only when data is not yet loaded', async () => {
    const component = mount(
      <MockedProvider>
        <ServiceBindingsWrapper lambdaName={lambda.name} />
      </MockedProvider>,
    );
    expect(component.find(Spinner)).toHaveLength(1);
    await componentUpdate(component);
    expect(component.find(Spinner)).toHaveLength(0);
  });
});
