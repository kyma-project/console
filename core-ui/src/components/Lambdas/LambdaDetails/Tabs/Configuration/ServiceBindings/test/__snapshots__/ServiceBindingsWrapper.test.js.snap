// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceBindingsWrapper Renders with an error in no lambdaName is provided 1`] = `
"Error! Network error: No more mocked responses for the query: query ServiceInstances($namespace: String!) {
  serviceInstances(namespace: $namespace) {
    name
    labels
    bindable
    status {
      type
      message
      __typename
    }
    servicePlan {
      bindingCreateParameterSchema
      __typename
    }
    serviceBindings {
      items {
        name
        parameters
        secret {
          name
          data
          __typename
        }
        status {
          type
          reason
          message
          __typename
        }
        __typename
      }
      __typename
    }
    serviceBindingUsages {
      name
      parameters {
        envPrefix {
          name
          __typename
        }
        __typename
      }
      serviceBinding {
        name
        secret {
          name
          data
          __typename
        }
        __typename
      }
      status {
        type
        reason
        message
        __typename
      }
      usedBy {
        name
        kind
        __typename
      }
      __typename
    }
    __typename
  }
}
, variables: {\\"namespace\\":\\"testnamespace\\"}"
`;

exports[`ServiceBindingsWrapper Renders with an error in no lambdaName is provided 2`] = `
"Warning: Failed prop type: The prop \`lambdaName\` is marked as required in \`ServiceBindingsWrapper\`, but its value is \`undefined\`.
    in ServiceBindingsWrapper (at ServiceBindingsWrapper.test.js:25)"
`;
